recipes/sherloc/inference/sherloc_utils/context_manager.py

```
@@ -0,0 +1,629 @@
+            if "<tool_call>" in gen_text and "</tool_call>" in gen_text:
+                # Extract the JSON between tool_call tags
+                start = gen_text.find("<tool_call>") + len("<tool_call>")
+                end = gen_text.find("</tool_call>")
+                if start < end:
+                    try:
+                        tool_call_json = gen_text[start:end].strip()
+                        # Normalize the JSON to handle formatting differences
+                        tool_call_obj = json.loads(tool_call_json)
+                        tool_call_normalized = json.dumps(tool_call_obj, sort_keys=True)
+                        tool_calls.append(tool_call_normalized)
+                    except (json.JSONDecodeError, Exception) as e:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)